packages/flame_yarn/lib/src/parse/token.dart

```diff
@@ -0,0 +1,147 @@
+import 'package:meta/meta.dart';
+
+/// [Token] is a unit of output during the lexing stage.
+@internal
+@immutable
+class Token {
+  const Token._(this.type, [this._content]);
+
+  const Token.command(String text) : this._(TokenType.command, text);
+  const Token.hashtag(String text) : this._(TokenType.hashtag, text);
+  const Token.id(String text) : this._(TokenType.id, text);
+  const Token.number(String text) : this._(TokenType.number, text);
+  const Token.speaker(String text) : this._(TokenType.speaker, text);
+  const Token.string(String text) : this._(TokenType.string, text);
+  const Token.text(String text) : this._(TokenType.text, text);
+  const Token.variable(String text) : this._(TokenType.variable, text);
+
+  static const arrow = Token._(TokenType.arrow);
+  static const as = Token._(TokenType.as);
```

Comment:
Will this work, isn't `as` a keyword?

Reply:
Apparently it does work, here's what the Dart spec says about it:
> (as, ...): These keywords are valid identifiers in most places, but they can’t be used as class or type names, or as import prefixes.

But perhaps we could rename it so as not to create unnecessary confusion.